### PR TITLE
chore(javascript): assert `authMode`

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1852

### Changes included:

while testing the client on the crawler I realized we had no tests for this js-specific option `authMode` and thought there was an issue, turns out it works well but at least now we have assertions for it

also add a browser suite for algoliasearch as we only run tests for node builds in the CTS

### CODE REVIEW

File: clients/algoliasearch-client-javascript/packages/algoliasearch/__tests__/algoliasearch.browser.test.ts [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    })) as unknown as EchoResponse;
+
+    expect(req.algoliaAgent).toMatchInlineSnapshot(
+      `"Algolia%20for%20JavaScript%20(${apiClientVersion})%3B%20Search%20(${apiClientVersion})%3B%20Browser"`
+    );
+  });
+
+  it('throws with undefined API key', () => {
+    try {
+      algoliasearch('APP_ID', '');
```
Comment (Fluf22) [inline clients/algoliasearch-client-javascript/packages/algoliasearch/__tests__/algoliasearch.browser.test.ts:37]: Shouldn't you put an expectError or something here, to be sure it throws and doesn't go past this line?

Comment (shortcuts) [reply clients/algoliasearch-client-javascript/packages/algoliasearch/__tests__/algoliasearch.browser.test.ts]: good point! I've went with https://github.com/algolia/api-clients-automation/pull/2897/commits/1e80b44d60387e68ea68c17bb76ad6260b11850b I believe that's more idiomatic

Comment (Fluf22) [reply clients/algoliasearch-client-javascript/packages/algoliasearch/__tests__/algoliasearch.browser.test.ts]: Very nice!

**Review by Fluf22 — APPROVED** [review]
🆗

**Pull request merged**